### PR TITLE
fix: ignore proposal window on vm version selection

### PR DESCRIPTION
**Pull request by driftluo**
### What problem does this PR solve?

Transaction submission in the current epoch can only use the VM version based on the current epoch 

### Check List

Tests

- Integration test in other repo

### Release note

```release-note
Title Only: Include only the PR title in the release note.
```

### CODE REVIEW

**Review by quake — APPROVED** [review]

File: script/src/verify.rs [hunk 1]
```diff
@@ -399,8 +399,7 @@ impl<'a, DL: CellDataProvider + HeaderProvider> TransactionScriptsVerifier<'a, D
 
     /// Returns the version of the machine based on the script and the consensus rules.
     pub fn select_version(&self, script: &'a Script) -> Result<ScriptVersion, ScriptError> {
-        let proposal_window = self.consensus.tx_proposal_window();
-        let epoch_number = self.tx_env.epoch_number(proposal_window);
+        let epoch_number = self.tx_env.current_epoch_number();
```
Comment (keroro520) [inline script/src/verify.rs:402]: Better add a comment explaining why we ignore the proposal window when selecting the VM version.

**Review by keroro520 — APPROVED** [review]

**Comment by keroro520** [comment]
Upon merging this PR, I will update https://github.com/nervosnetwork/ckb-integration-test/pull/13 to reflect this change.

**Comment by keroro520** [comment]
bors r=quake,keroro520

**Comment by driftluo** [comment]
bors r=quake,keroro520

**Comment by driftluo** [comment]
bors r=quake,keroro520

**Comment by keroro520** [comment]
bors r=quake,keroro520

**Pull request merged**